clevertap-core/src/main/java/com/clevertap/android/sdk/inapp/images/InAppResourceProvider.kt

```diff
@@ -4,83 +4,97 @@ import android.content.Context
+
     companion object {
+
         private const val IMAGE_DIRECTORY_NAME = "CleverTap.Images."
         private const val GIF_DIRECTORY_NAME = "CleverTap.Gif."
+        private const val ALL_FILE_TYPES_DIRECTORY_NAME = "CleverTap.Files."
     }
 
     fun saveImage(cacheKey: String, bitmap: Bitmap, bytes: ByteArray) {
-
-        val imageMemoryCache = ctCaches.imageCache()
-        imageMemoryCache.add(cacheKey, bitmap)
-
-        val imageDiskCache = ctCaches.imageCacheDisk(dir = images)
-        imageDiskCache.add(cacheKey, bytes)
+        val imageMAO = ImageMemoryAccessObject(ctCaches)
+        val savedFile = imageMAO.saveDiskMemory(cacheKey, bytes)
+        imageMAO.saveInMemory(cacheKey, Pair(bitmap, savedFile))
     }
 
     fun saveGif(cacheKey: String, bytes: ByteArray) {
-        val gifMemoryCache = ctCaches.gifCache()
-        gifMemoryCache.add(cacheKey, bytes)
-
-        val gifDiskCache = ctCaches.gifCacheDisk(dir = gifs)
-        gifDiskCache.add(cacheKey, bytes)
+        val gifMAO = GifMemoryAccessObject(ctCaches)
+        val savedFile = gifMAO.saveDiskMemory(cacheKey, bytes)
+        gifMAO.saveInMemory(cacheKey, Pair(bytes, savedFile))
     }
 
-    fun isImageCached(url: String) : Boolean {
-        val imageMemoryCache = ctCaches.imageCache()
-
-        if (imageMemoryCache.get(url) != null) {
-            return true
-        }
+    fun saveFile(cacheKey: String, bytes: ByteArray) {
+        val fileMAO = FileMemoryAccessObject(ctCaches)
+        val savedFile = fileMAO.saveDiskMemory(cacheKey, bytes)
+        fileMAO.saveInMemory(cacheKey, Pair(bytes, savedFile))
+    }
 
-        val imageDiskCache = ctCaches.imageCacheDisk(dir = images)
-        val file = imageDiskCache.get(url)
+    fun isImageCached(url: String): Boolean {
+        return isFileCached(url)
+    }
 
-        return (file != null)
+    fun isGifCached(url: String): Boolean {
+        return isFileCached(url)
     }
 
-    fun isGifCached(url: String) : Boolean {
-        val gifMemoryCache = ctCaches.gifCache()
+    fun isFileCached(url: String): Boolean {
+        val memoryAccessObjectList = listOf<MemoryAccessObject<*>>(
+            FileMemoryAccessObject(ctCaches), ImageMemoryAccessObject(ctCaches),
+            GifMemoryAccessObject(ctCaches)
+        )
 
-        if (gifMemoryCache.get(url) != null) {
-            return true
+        // Try in memory
+        memoryAccessObjectList.forEach {
+            val pair = it.fetchInMemory(url)
+            if (pair != null) {
+                return true
+            }
         }
 
-        val gifDiskCache = ctCaches.gifCacheDisk(dir = images)
-        val file = gifDiskCache.get(url)
-
-        return (file != null)
+        // Try disk
+        memoryAccessObjectList.forEach {
+            val file = it.fetchDiskMemory(url)
+            if (file != null) {
+                return true
+            }
+        }
+        return false
     }
 
     fun cachedImage(cacheKey: String?): Bitmap? {
```

Comment:
Can we rename to getCachedImage. Would be more readable?

Reply:
there is no equivalent method that starts with prefix `set` let's keep as it is. https://stackoverflow.com/a/10568713